cirq-core/cirq/study/sweeps.py

```diff
@@ -483,4 +486,9 @@ def dict_to_zip_sweep(factor_dict: ProductOrZipSweepLike) -> Zip:
     Returns:
         Zip product of the sweeps.
     """
-    return Zip(*(Points(k, v if isinstance(v, Sequence) else [v]) for k, v in factor_dict.items()))
+    return Zip(
+        *(
+            Points(k, cast(float, v) if isinstance(v, Sequence) else [v])  # type: ignore
```

Comment:
nit: add mypy error code

Reply:
Same thing, multiple mypy error codes since it is in a list comprehension.